### DISCUSSION TITLE
[Fabric] How can I use react-native-vector-icons in the new cpp-template?

### DISCUSSION THREAD

**Discussion by l3utterfly** [microsoft/react-native-windows#12954] [Q&A] [answered]
I need to copy font files into the "assets" directory in visual studio, how do I do that?

**Comment by chrisglein** [comment]
The module doesn't claim to support windows in the directory: https://reactnative.directory/?search=react-native-vector-icons
And the module itself lists Window setup steps, and has a Windows example, but I don't see any Windows native code. Is this a pure JS module or is there native code? I'm not familiar with this one.

But is this your primary question?
> I need to copy font files into the "assets" directory in visual studio, how do I do that?

If you copy the files via the file system then you don't need to do any setup in visual studio. The bundler should pick up anything in there.

Can you provide more context to your question?

**Reply by l3utterfly** [reply]
I have some font files I need to reference in the JS code to display icons.

Where should I edit in the bundler settings to include fonts?

If react-native-vector-icons doesn't work, I'm happy to implement it manually.

**Reply by chrisglein** [reply, marked as answer]
If you put files in the assets folder they will be copied over for the bundle. The question is how do you refer to them. Nothing about putting them in the bundle will have them automatically loaded as valid font faces. So there needs to be some Uri to the loaded file. If you poke around in the vector-icons you can see some per-platform logic there wit adding the files. I'm not actually positive what the steps are for UWP fonts (which is what you'd need for RNW Paper). There are some articles on it like [this](https://www.c-sharpcorner.com/article/custom-fonts-in-windows-10-uwp-app/). There might be something from within the main [Windows typography](https://learn.microsoft.com/en-us/windows/apps/design/style/typography) articles, I'm not sure.  Not saying it isn't possible, I'm just not personally familiar. I do see that [Xamarin has some detail on this](https://devblogs.microsoft.com/xamarin/embedded-fonts-xamarin-forms/), which might have some nuggets of how that might carry over.

Not sure if it helps you, but the [Segoe Icon Font](https://learn.microsoft.com/en-us/windows/apps/design/style/segoe-ui-symbol-font) is _really_ helpful if you want that sort of thing. For adding icons that match the Windows Fluent design language, I've found that invaluable. And since it's just a font that'll work great on either Paper or Fabric RNW.
